.github/copilot-instructions.md

```diff
@@ -0,0 +1,194 @@
+
+**Note:** The README mentions "./gradlew build needs to run twice to work" but this was not observed during testing.
+
+## Project Layout and Architecture
+
+### Module Structure
+```
+gradle-common/
+├── kotlin-common-gradle-plugins/     # Core Kotlin plugin conventions
+├── architecture-common-gradle-plugins/ # Web/architecture-specific plugins
+├── common-gradle-dependencies/       # Shared dependency versions
+├── huanshankeji-team-gradle-plugins/ # Team-specific utilities
+├── buildSrc/                        # Build configuration and scripts
+├── .github/workflows/               # CI/CD pipelines
+└── gradle/wrapper/                  # Gradle wrapper
+```
+
+### Key Configuration Files
+
+**Root Level:**
+- `build.gradle.kts` - Root build configuration with Dokka and plugin publishing
+- `settings.gradle.kts` - Module inclusion and dependency resolution
+- `gradle.properties` - Build properties (JVM args, Dokka settings)
+
+**Build Configuration:**
+- `buildSrc/src/main/kotlin/VersionsAndDependencies.kt` - Centralized version management
+- `buildSrc/src/main/kotlin/Constants.kt` - Project constants
+- `buildSrc/build.gradle.kts` - Build dependencies and Kotlin version
+
+**CI/CD:**
+- `.github/workflows/kotlin-jvm-ci.yml` - Main CI pipeline (Ubuntu, JDK 8)
+- `.github/workflows/dokka-gh-pages.yml` - Documentation deployment
+- `.github/workflows/copilot-setup-steps.yml` - Setup automation
+
+### Major Components
+
+**kotlin-common-gradle-plugins:** Convention plugins for Kotlin projects including:
+- `kotlin-jvm-library-*` plugins for JVM libraries
+- `kotlin-multiplatform-*` plugins for multiplatform projects
+- Maven publishing conventions
+- Sonatype OSSRH publishing
+
+**architecture-common-gradle-plugins:** Web and architecture plugins including:
+- `GenerateKotlinJsBrowserWebrootForVertxWebPlugin` for Vert.x integration
+- Web frontend conventions
+- JavaScript browser distribution handling
+
+**common-gradle-dependencies:** Centralized dependency versions in `CommonVersions.kt`
+
+### Validation Pipeline
+
+The repository uses GitHub Actions CI that:
+1. Runs on JDK 8 with Temurin distribution
+2. Executes `gradle-test-and-check` action
+3. Performs dependency submission for security scanning
+4. Generates API documentation with Dokka
+
+**To replicate CI locally:**
+```bash
+./gradlew check                    # Equivalent to CI test-and-check
+./gradlew publishToMavenLocal      # Verify publishing works
+./gradlew apiCheck                 # Verify API compatibility
+```
+
+### Dependencies and Compatibility
+
+**External Dependencies:**
+- Kotlin Gradle Plugin 2.2.20
+- Compose Multiplatform 1.10.0-alpha01
+- Dokka 2.1.0-Beta
+- Gradle Plugin Publish Plugin 1.3.0
+- Kotlin Binary Compatibility Validator 0.18.1
+
+**Version Compatibility Notes:**
+- Gradle 9.0.0 required (handled by wrapper)
+- Kotlin version conflicts may occur - exclude transitive Kotlin dependencies if needed
```

Comment:
No "exclude transitive Kotlin dependencies" from README is for building consuming projects not for building this project. "Kotlin version conflicts" are normal when the Kotlin Gradle plugin version is newer that the version bundled in Gradle.